# SCTLD TAG-seq Jamboree  

## Introduction  
### General Overview

Here you will find the scripts and pipeline for the analysis of the TAG-seq (3 prime RNA-seq) analysis of *Orbicella faveolata* and *Montastrea cavernosa*. This workflow assumes basic understanding of Unix bash and R languages, but following tutorials for each step should provide enough background.  

These samples were sequenced on a Nova-seq (at the University of Miami Center for Genome Technology Core Facility) with single-end 100 base pair reads. 

TAG-seq employs a very similar pipeline to traditional RNA-seq with a few tweaks due to how library prep is performed. This is explained briefly below. 

![alt text here](./example_figures/RNAseq_overview.png)

### General RNA-seq analysis Overview


## TagSeq
  Scripts for general TagSeq analyses provided by Matz Lab at UT Arlington:
  https://github.com/z0on/tag-based_RNAseq
  
  Helpful slides higlighting difference between TagSeq and standard RNAseq:
  https://ucdavis-bioinformatics-training.github.io/2017-June-RNA-Seq-Workshop/wednesday/TagSeqVSRnaSeq.pdf

    *Montastrea cavernosa*
  Genome resources from the Matz lab: https://matzlab.weebly.com/data--code.html
    
    *Orbicella faveolata*
  Genome resources from the NCBI: https://www.ncbi.nlm.nih.gov/assembly/GCF_002042975.1/

## RNAseq

  