---
title: "sctld_jamboree_final_analysis"
author: "Allyson DeMerlis"
date: "2/15/2021"
output: html_document
---

###Libraries 

Install and Load Packages
```{r}
library(tidyverse)
library(DESeq2)
library(ggrepel)
library(apeglm)
library(EnhancedVolcano)
library(ComplexHeatmap)
library(pheatmap)
library(genefilter)
library(WGCNA)
library(flashClust)
library(edgeR) #counts per million
library(readxl)
library(here)
```

### Importing raw counts 

Import counts data for orbicella faveolata from feature counts (removed low read depth samples and kept only Mote samples)
```{r}
ofav_counts <- read.delim(here("data tables and spreadsheets/ofav_counts"), row.names = 1, skip = 1)

# Project Species Status n
# Mote	Ofav	control	  19	
# Mote	Ofav	exposed	  19	
# Smithsonian	Ofav	control	3	
# Smithsonian	Ofav	exposed	6	

#reformatting the counts file for use
colnames(ofav_counts) <- gsub("X.scratch.projects.transcriptomics.mikeconnelly.projects.sctld_jamboree.tagseq.outputs.alignments.Ofav_na_", "", colnames(ofav_counts))
colnames(ofav_counts) <- gsub("_na_Aligned.sortedByCoord.out.uniq.bam", "", colnames(ofav_counts))
colnames(ofav_counts) <- gsub("_", "", colnames(ofav_counts))
colnames(ofav_counts) <- gsub("na", "", colnames(ofav_counts))

ofav_counts %>%select(!Chr:Length) -> ofav_counts
```

Import Montastrea cavernosa counts 
```{r}
mcav_counts <- read.table(here("data tables and spreadsheets/mcav_counts"), header = T) %>% 
  column_to_rownames(var = "Geneid") %>% 
  select(-Chr, -Start, -End, -Strand, -Length) -> mcav_counts #17 samples
  
# Project Species Status n
# Mote	Mcav	control   17	
# Mote	Mcav	exposed	  18	
# Smithsonian	Mcav	control	3	
# Smithsonian	Mcav	exposed	4	
#Total samples = 42

colnames(mcav_counts) <- gsub("Aligned.sortedByCoord.out.bam_unique.bam", "", colnames(mcav_counts))
colnames(mcav_counts) <- gsub("Mcav_na_", "", colnames(mcav_counts))
colnames(mcav_counts) <- gsub("_", "", colnames(mcav_counts))
colnames(mcav_counts) <- gsub("na", "", colnames(mcav_counts))
```

Subset Ofav samples to remove SMS samples and 5 Mote diseased samples that were clustering with controls
```{r}
ofav_counts %>% 
  select(-K76D1, -K80D1, -K82D1, -K78D1, -K74D1) -> ofav_mote_subset

#write_csv(ofav_mote_subset, "ofav_mote_subset_ofav_rawcounts.csv")
```

### Metadata files 

create Ofav metadata file
```{r}
#making the metadata file for the sample and treatment information
ofav_metadata=data.frame(sample = colnames(ofav_mote_subset),
                condition = stringr::str_detect(pattern = "control",string = colnames(ofav_mote_subset))
                )
#changing TRUE and FALSE for condition to Control and Wounded
ofav_metadata$condition[str_detect(ofav_metadata$condition,"TRUE")] <- "Control"
ofav_metadata$condition[str_detect(ofav_metadata$condition,"FALSE")] <- "Diseased"

ofav_metadata<-ofav_metadata %>% column_to_rownames("sample")

ncol(ofav_mote_subset) == nrow(ofav_metadata)

ofav_counts_ofav_matrix <- as.matrix(ofav_mote_subset)
```

Mcav metadata with both SMS and Mote
```{r}
mcav_metadata=data.frame(sample = colnames(mcav_counts),
                condition = stringr::str_detect(pattern = "control",string = colnames(mcav_counts))
                )

mcav_metadata$condition[str_detect(mcav_metadata$condition,"TRUE")] <- "Control"
mcav_metadata$condition[str_detect(mcav_metadata$condition,"FALSE")] <- "Diseased"
mcav_metadata<-mcav_metadata %>% column_to_rownames("sample")

ncol(mcav_counts) == nrow(mcav_metadata)

countdata_mcav_matrix <- as.matrix(mcav_counts)
```



###Create DESeq Objects
```{r filtering}
dds_ofav <- DESeqDataSetFromMatrix(countData=ofav_counts_ofav_matrix, colData=ofav_metadata, design= ~condition)

keep2<- rowSums(counts(dds_ofav)>=4) >=5 # filtering less than 4 samples with counts less than or equal to 5
dds_ofav <- dds_ofav[keep2,]

nrow(dds_ofav) #20195 genes

dds_mcav <- DESeqDataSetFromMatrix(countData=countdata_mcav_matrix, colData=mcav_metadata, design= ~condition)
nrow(dds_mcav) #24740 genes

keep<- rowSums(counts(dds_mcav)>=4) >=5
dds_mcav <- dds_mcav[keep, ]
```

### VST Normalization and PCA plotting
```{r}
dds_vst_ofav <- vst(dds_ofav, blind = T) #use blind = TRUE to not account for experimental design

plotPCA(dds_vst_ofav)

dds_vst_mcav <- vst(dds_mcav, blind = T)

plotPCA(dds_vst_mcav)
```

### PCA GGplots
Ofav PCA axes 1 and 2
```{r}
pca12 <- plotPCA(dds_vst_ofav,intgroup=c("condition"),returnData = TRUE)
ggplot(pca12, aes(PC1, PC2,color=condition)) + 
  geom_point(size=3) +  xlab(paste0("PC1 44% variance")) + 
  ylab(paste0("PC2 12% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 3) +
  theme_classic() + 
  stat_ellipse(aes(PC1, PC2, group=condition), type = "norm") +
  scale_color_manual(values = (c("#7CBAF5", "#AD161A")))
```
Ofav PCA axes 2 and 3
```{r Pretty PC2 and PC3}
#PCA 2 and 3 axis creation
pcaaxes23 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC2 = pca$x[, 2], PC3 = pca$x[, 3], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[2:3]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC2", y = "PC3", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC2: ", round(percentVar[2] * 
        100), "% variance")) + ylab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + coord_fixed()
}
pcaaxes23(dds_vst_ofav,
          intgroup = c("condition"),
          returnData = F)

pca23 <-
  pcaaxes23(dds_vst_ofav,
          intgroup = c("condition"),
          returnData = TRUE)

ggplot(pca23, aes(PC2, PC3, color = condition)) +
  geom_point(size = 3) +  xlab(paste0("PC2 12% variance")) +
  ylab(paste0("PC3 9% variance")) +
  theme(legend.position = "right")  +
  theme(text = element_text(size = 10))  +
  theme(legend.key.size = unit(0.5, "cm")) +
  geom_point(size = 3) +
  theme_classic() +
  stat_ellipse(aes(PC2, PC3, group = condition), type = "norm") +
  scale_color_manual(values = (c("#7CBAF5", "#AD161A")))
```
Ofav PCA Axes 3 and 4
```{r Pretty PC3 and PC4}
#PCA 3 and 4 axis creation
pcaaxes34 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC3 = pca$x[, 3], PC4 = pca$x[, 4], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[3:4]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC3", y = "PC4", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + ylab(paste0("PC4: ", round(percentVar[4] * 
        100), "% variance")) + coord_fixed()
}

pcaaxes34(dds_vst_ofav,
          intgroup = c("condition"),
          returnData = F)

pca34 <-
  pcaaxes34(dds_vst_ofav,
          intgroup = c("condition"),
          returnData = TRUE)

ggplot(pca34, aes(PC3, PC4, color = condition)) +
  geom_point(size = 3) +  xlab(paste0("PC3 9% variance")) +
  ylab(paste0("PC4 6% variance")) +
  theme(legend.position = "right")  +
  theme(text = element_text(size = 10))  +
  theme(legend.key.size = unit(0.5, "cm")) +
  geom_point(size = 3) +
  theme_classic() +
  stat_ellipse(aes(PC3, PC4, group = condition), type = "norm") +
  scale_color_manual(values = (c("#7CBAF5", "#AD161A")))
```

Mcav ggplot PCA Axes 1 and 2 
```{r}
pca12_mcav <- plotPCA(dds_vst_mcav,intgroup=c("condition"),returnData = TRUE)
ggplot(pca12_mcav, aes(PC1, PC2,color=condition)) + 
  geom_point(size=3) +  xlab(paste0("PC1 20% variance")) + 
  ylab(paste0("PC2 17% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 3) +
  theme_classic() + 
  stat_ellipse(aes(PC1, PC2, group=condition), type = "norm") +
  scale_color_manual(values = (c("#7CBAF5", "#AD161A")))
```

### DESeq Analysis
```{r}
dds_ofav <- DESeq(dds_ofav)
res_ofav <- results(dds_ofav, alpha = 0.05)
summary(res_ofav) # 4746 DEGs, 20195 genes, 0 outliers, 784 low counts

dds_mcav <- DESeq(dds_mcav)
res_mcav <- results(dds_mcav, alpha = 0.05)
summary(res_mcav) #316 DEGs, 9210 genes, 1096 outliers, 2170 low counts
```

### Protein Feature Annotations to Results Tables
```{r}
res_ofav %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Locus") -> ofav_results_table

write_csv(ofav_results_table, "ofav_results_table.csv")

mcav_results_table <- (as.data.frame(res_mcav)) 
mcav_results_table %>% rownames_to_column("transcript_id") -> mcav_results_table

write_csv(mcav_results_table, "mcav_results_table.csv")

ofav_prot <- read.csv(here("data tables and spreadsheets/old_Ofav_Mote/ofav_proteins_13173_311351.csv")) #this table is linking gene IDs to protein IDs and protein names.
mcav_prot <- read_csv(here("data tables and spreadsheets/mcav_annot.csv"))
```

EggNOG
```{r}
eggNOGcols <-
  c(
    "Query",
    "seed_eggNOG_ortholog",
    "seed_ortholog_evalue",
    "seed_ortholog_score",
    "best_tax_level",
    "Preferred_name",
    "GOs",
    "EC",
    "KEGG_ko",
    "KEGG_Pathway",
    "KEGG_Module",
    "KEGG_Reaction",
    "KEGG_rclass",
    "BRITE",
    "KEGG_TC",
    "CAZy",
    "BiGG_Reaction",
    "annot lvl",
    "matching OGs",
    "Best OG",
    "COG_cat",
    "description"
  )

ofav_eggnog <- read.delim(here("data tables and spreadsheets/old_Ofav_Mote/ofav.protein.emapper.annotations.csv"), 
                          comment.char  = "#", 
                          header = F, 
                          col.names = eggNOGcols) 
#this file contains the full annotations from the Ofav proteins (KOG/GO terms, PFAM domains, etc, obtained from eggnog-mapper.embl.de)

mcav_eggnog <- read.delim(here("data tables and spreadsheets/mcav.protein.emapper.annotations"), comment.char  = "#", header = F,  col.names = eggNOGcols)

ofav_eggnog %>% 
  dplyr::rename(Protein.product = Query) -> ofav_eggnog

ofav_annot_DEG <- left_join(ofav_results_table, ofav_prot, by="Locus")

#check these tables before uploading them to basecamp
ofav_annot_DEG_emap <- left_join(ofav_annot_DEG, ofav_eggnog, by="Protein.product")

mcav_eggnog %>% dplyr::rename(geneid = Query) -> mcav_eggnog

mcav_annot_DEG <- left_join(mcav_results_table, mcav_prot, by="transcript_id")
```

edit mcav DEG annotation so that protein name comes from SPROT top BLASTX hit
```{r}
mcav_annot_DEG %>% 
  select(transcript_id:padj, sprot_Top_BLASTX_hit) %>% 
  dplyr::rename(Locus = transcript_id) %>% 
  separate(sprot_Top_BLASTX_hit, sep = "Full=", into = c("before", "Protein.name")) %>% 
  select(!before) %>% 
  separate(Protein.name, sep = "/", into = "Protein.name") %>% 
  separate(Protein.name, sep = ";", into = "Protein.name") 
 #write.csv("mcav_annot_DEG_proteinnames.csv")
```

Comparing DEG list that Ben got versus what we got initially
```{r}
updated_mcav_DEG_annotated <- read_csv("mcav_DEG_annotated_updated.csv")

length(unique(mcav_annot_DEG$transcript_id)) #9210
length(unique(updated_mcav_DEG_annotated$X1)) #582

original_gene_list <- mcav_annot_DEG$transcript_id
updated_gene_list <-updated_mcav_DEG_annotated$X1

length(intersect(updated_gene_list, original_gene_list)) #538 genes

updated_ofav_DEG_annotated <- read_csv("ofav_DEG_annotated_updated.csv")

length(unique(ofav_annot_DEG_emap$Locus)) #20195
length(unique(updated_ofav_DEG_annotated$X1)) #2194

original_gene_list_ofav <- ofav_annot_DEG_emap$Locus
updated_gene_list_ofav <-updated_ofav_DEG_annotated$X1

length(intersect(updated_gene_list_ofav, original_gene_list_ofav)) #2189 genes

```


### Volcano Plots with DEGs
Ofav
```{r Volcano Plot, fig.height=4, fig.width=3}
EnhancedVolcano(res_ofav,
    lab = NA,
    x = 'log2FoldChange',
    y = 'pvalue',
    col=c('grey', 'black', 'blue', 'purple'),
    pCutoff = 0.05,
    FCcutoff = 1)
ggsave("volcanoplot_ofav_05.png", height = 12, width = 10)
#The default cut-off for log2FC is >|2|; the default cut-off for P value is 10e-6. 
```

Mcav
```{r}
EnhancedVolcano(res_mcav,
    lab = NA,
    x = 'log2FoldChange',
    y = 'pvalue',
    col=c('grey', 'black', 'blue', 'purple'), pCutoff = 0.05,
    FCcutoff = 1)

 ggsave("volcanoplot_mcav_05.png", height = 12, width = 10)
#The default cut-off for log2FC is >|2|; the default cut-off for P value is 10e-6. 
```



### Heatmaps of Count Matrix
```{r}
select_ofav <- order(rowMeans(counts(dds_ofav,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df_ofav <- ofav_metadata

#svg("mcav_vst_heatmap.svg")
pheatmap(assay(dds_vst_ofav)[select_ofav,], cluster_rows=FALSE, #using vst normalization
          annotation_col=df_ofav)


select_mcav <- order(rowMeans(counts(dds_mcav,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df_mcav <- mcav_metadata

#svg("mcav_vst_heatmap.svg")
pheatmap(assay(dds_vst_mcav)[select_mcav,], cluster_rows=FALSE, #using vst normalization
          annotation_col=df_mcav)

#these heatmaps were then exported into Adobe Illustrator, where Allyson manually edited the Loci names to be the protein names from the annotated files.
```

